### PR TITLE
fix bugs and add path and git option

### PR DESCRIPTION
**Pull request by masterDev32**

### CODE REVIEW

**Comment by masterDev32** [comment]
@Nesci28 can you test it in your side if everything is ok?

File: app.js [hunk 1]
```diff
@@ -133,50 +133,55 @@ const tslint = {
...
       filter: t => {
-        return t.replace(/[^a-zA-Z]/g, '');
-      },
+        return t.replace(/[^a-zA-Z-]/g, '');
+      }
+    },
+    {
+      type: 'text',
+      name: 'path',
+      message: 'Do you want a specific path? if so please specify it: \n'
```
Comment (Nesci28) [inline app.js]: Typo and missing an emoji to fit with the other questions.

Please use: https://www.npmjs.com/package/inquirer-select-directory
to select a path.  That way, no bug from the end-user can happen in the path selection.

File: app.js [hunk 2]
```diff
@@ -235,31 +240,52 @@ const tslint = {
...
+      name: 'git',
+      message: 'Do you want to add a git repositorie ? :octocat: ',
+      default: false
     },
+    {
+      type: 'text',
+      name: 'gitLink',
+      message: 'Provide a git repositorie link(ended by .git)  ',
+      when: answers => {
+        return answers.git === true;
```
Comment (Nesci28) [inline app.js]: === true is redundant
the when attribute happens when it returns true.  answers.git is already a boolean

File: app.js [hunk 3]
```diff
@@ -276,16 +302,24 @@ const tslint = {
...
-    printMsg('Installing NPX...');
-    cp.execSync(`npm install -g npx > ${toNull}`);
-    printDone('Installing NPX...');
-    printMsg('Creating the Angular App ...');
-    cp.execSync(
-      `npx -p @angular/cli@${answers.version} ${angularNewCommand} > ${toNull}`,
-    );
+    let version;
+    try {
+      version = cp.execSync(`npx -version`);
```
Comment (Nesci28) [inline app.js]: add .toString() here because cp always return a Buffer.

File: app.js [hunk 4]
```diff
@@ -276,16 +302,24 @@ const tslint = {
...
+    } catch (error) {
+      printMsg('Installing NPX...');
+      cp.execSync(`npm install -g npx > ${toNull}`);
+      printDone('Installing NPX...');
+    }
+    if (version && !!version.toString()) {
```
Comment (Nesci28) [inline app.js]: Since version will be converted to a string, the !!version.toString() becomes redundant

File: app.js [hunk 5]
```diff
@@ -410,24 +442,42 @@ const tslint = {
...
   }
 
+  if (answers.git) {
+    printMsg('Linkining to git...');
+    try {
+      version = cp.execSync(`git --version`);
+    } catch (error) {
+      printDone('git is not installed...');
+    }
+    if (version && !!version.toString()) {
```
Comment (Nesci28) [inline app.js]: Same comment for the !!version.toString()

File: app.js [hunk 6]
```diff
@@ -410,24 +442,42 @@ const tslint = {
     printDone('Installing HammerJS...');
   }
 
+  if (answers.git) {
+    printMsg('Linkining to git...');
+    try {
```
Comment (Nesci28) [inline app.js:449]: I like the idea here.
But that could be in the "when" property in the inquirer.prompt so the question wont even be asked if Git is not installed

File: app.js [hunk 7]
```diff
@@ -410,24 +442,42 @@ const tslint = {
...
+      version = cp.execSync(`git --version`);
+    } catch (error) {
+      printDone('git is not installed...');
+    }
+    if (version && !!version.toString()) {
+      cp.execSync(`git init`);
```
Comment (Nesci28) [inline app.js]: ng new already creates a git repo by default, but we could add a question if you want to generate a git repo when creating the angular app
https://angular.io/cli/new
--commit=true|false

File: app.js [hunk 8]
```diff
@@ -410,24 +442,42 @@ const tslint = {
...
+      fs.writeFile('./.gitignore', 'node_modules', () => {});
```
Comment (Nesci28) [inline app.js]: same comment, angular creates a .gitignore file by default which is already packed with nodejs and angular stuff to ignore

**Review by Nesci28 — CHANGES_REQUESTED** [review]

**Pull request merged**